src/client/src/App/Credit/SellSide/sellSideState.ts

```diff
@@ -0,0 +1,144 @@
+import { combineLatest, merge } from "rxjs"
+import { map, startWith, delay } from "rxjs/operators"
+import { bind } from "@react-rxjs/core"
+import { createSignal } from "@react-rxjs/utils"
+import { QuoteState, RfqState } from "@/generated/TradingGateway"
+import {
+  ADAPTIVE_BANK_NAME,
+  creditRfqsById$,
+  RfqDetails,
+} from "@/services/credit"
+import { invertDirection } from "@/utils"
+import { timeRemainingComparator } from "../common"
+import { RfqRow } from "./SellSideRfqGrid"
+import { HIGHLIGHT_ROW_FLASH_TIME } from "@/constants"
+
+export enum SellSideQuoteState {
+  New = "New RFQ",
+  Pending = "Quote Sent",
+  Expired = "Expired",
+  Cancelled = "Cancelled",
+  Lost = "Traded Away",
+  Rejected = "Rejected",
+  Passed = "Passed",
+  Accepted = "Done",
+}
+
+export const getSellSideQuoteState = (
+  rfqState: RfqState,
+  quoteState: QuoteState | undefined,
+): SellSideQuoteState => {
+  if (rfqState === RfqState.Cancelled) {
+    return SellSideQuoteState.Cancelled
+  } else if (rfqState === RfqState.Expired) {
+    return SellSideQuoteState.Expired
+  } else if (rfqState === RfqState.Open && quoteState === undefined) {
+    return SellSideQuoteState.New
+  } else if (
+    rfqState === RfqState.Closed &&
+    quoteState !== QuoteState.Accepted
+  ) {
+    return SellSideQuoteState.Lost
+  } else if (rfqState === RfqState.Open && quoteState === QuoteState.Pending) {
+    return SellSideQuoteState.Pending
+  } else if (quoteState === QuoteState.Rejected) {
+    return SellSideQuoteState.Rejected
+  } else if (quoteState === QuoteState.Accepted) {
+    return SellSideQuoteState.Accepted
+  } else {
+    throw new Error()
+  }
+}
+
+const [_highlightRfqId$, highlightRfqId] = createSignal<number | null>()
+export { highlightRfqId }
+
+export const [useHighlightedRfqId] = bind(
+  merge(
+    _highlightRfqId$,
+    _highlightRfqId$.pipe(
+      map(() => null),
+      delay(HIGHLIGHT_ROW_FLASH_TIME),
+    ),
+  ),
+  null,
+)
+
+const [_$, selectRfqId] = createSignal<number | null>()
+export { selectRfqId }
+export const [useSelectedRfqId, selectedRfqId$] = bind(_$, null)
+
+export enum SellSideQuotesTab {
+  All = "All RFQs",
+  Live = "Live RFQs",
+  Closed = "Closed RFQs",
+}
+
+export const SELLSIDE_RFQS_TABS: SellSideQuotesTab[] =
+  Object.values(SellSideQuotesTab)
+
+const [___$, setQuotesFilter] = createSignal<SellSideQuotesTab>()
```

Comment:
not a fan of the underscorey names .. is this to avoid further naming decisions for immediate internally used streams?

Reply:
Yeah, it got silly.  I'll push up a slightly more verbose alternative.  It's just to signal that it's a value that only exists to be input immediately into bind.  And, because what bind produces in the second element of the tuple is also a similar Observable (one that is shared/multicast), they can get confused.